probatus/sample_similarity/resemblance_model.py

```diff
@@ -99,38 +107,31 @@ def fit(self, X1, X2, column_names=None):
         np.random.seed(self.random_state)
 
         # Ensure inputs are correct
-        self.X1 = assure_numpy_array(X1)
-        self.X2 = assure_numpy_array(X2)
-
-        # Check if any missing values
-        warn_if_missing(self.X1, 'X1')
-        warn_if_missing(self.X2, 'X2')
-
-        # Ensure the same shapes
-        if self.X1.shape[1] != self.X2.shape[1]:
-            raise(ValueError("Passed variables do not have the same shape. The passed dimensions are {} and {}".
-                             format(self.X1.shape[1], self.X2.shape[1])))
-
-        # Check if column_names are passed correctly
-        self.column_names = assure_column_names_consistency(column_names, X1)
+        self.X1, self.column_names = preprocess_data(X1, X_name='X1', column_names=column_names, verbose=self.verbose)
+        self.X2, _ = preprocess_data(X2, X_name='X2', column_names=column_names, verbose=self.verbose)
 
         # Prepare dataset for modelling
-        self.X = pd.DataFrame(np.concatenate([
+        self.X = pd.DataFrame(pd.concat([
             self.X1,
             self.X2
-        ]), columns = self.column_names).reset_index(drop=True)
+        ], axis=0), columns = self.column_names).reset_index(drop=True)
 
         self.y = pd.Series(np.concatenate([
             np.zeros(self.X1.shape[0]),
             np.ones(self.X2.shape[0]),
         ])).reset_index(drop=True)
 
+        # Assure the type and number of classes for the variable
+        self.X, _ = preprocess_data(self.X, X_name='X', column_names=self.column_names, verbose=self.verbose)
+
+        self.y = preprocess_labels(self.y, y_name='y', index=self.X.index, verbose=self.verbose)
+
         # Reinitialize variables in case of multiple times being fit
         self._init_output_variables()
 
-        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(self.X, self.y,test_size=self.test_prc,
+        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(self.X, self.y, test_size=self.test_prc,
                                                                                 random_state=self.random_state,
-                                                                                stratify=self.y)
+                                                                                shuffle=True, stratify=self.y)
```

Comment:
good catch. small change, good to make a list of all these small tweaks in the changelog ;)